website/pages/docs/theming/advanced.mdx

```diff
@@ -97,3 +97,44 @@ interface StyleConfig {
   }
 }
 ```
+
+## Theme Typings
+
+<Badge fontSize="sm" colorScheme="teal" letterSpacing="wider">
+  v1.2.0
+</Badge>
+
+When adding new components, component variations, sizes, colors and other theme
+foundations, you can extend the internal theme typings to provide autocomplete
+for your application theme.
+
+If you want to learn how to scale your custom theme you can
+[follow this guide](/docs/theming/customize-theme#scaling-out-your-project).
+
+```bash
+npx @chakra-ui/cli tokens <path/to/your/theme.(js|ts)>
+```
+
+The theme entrypoint file should export the theme object either as `default`
+export or as named `theme` export.
+
+> Note 🚨: If you delete the `node_modules` directory, you'll need to re-run the
+> command to get proper typings again.
+
+For convenience, you can add a `postinstall` script to your `package.json`, so you
+don't have to think about this every time you reinstall your dependencies.
+
+```json
+"scripts": {
+  "gen:theme-typings": "chakra-cli tokens <path/to/your/theme.(js|ts)>",
+  "postinstall": "npm run gen:theme-typings"
+}
+```
+
+The autocomplete for theme token **only works in _Visual Studio Code_ for now**.
+We keep an eye on the TypeScript support of Jetbrains IDEs, but I we have to
+wait until it is improved.
```

Comment:
```suggestion
We'll keep an eye on the TypeScript support of Jetbrains IDEs, but we have to
wait until it is improved.
```

Reply:
@TimKolberger Do you know if there's a specific issue on the JetBrains bug tracker that would enable us to support this? Would love to link to that if possible.

Reply:
There are many different issues with the TypeScript support of JetBrains IDEs.

I created a reproduction for required prop warnings [WEB-48161](https://youtrack.jetbrains.com/issue/WEB-48161) [here](https://gist.github.com/TimKolberger/0900962f7f2c28b4a80bcd8754785cf2).

Missing auto import of components here [WEB-48085](https://youtrack.jetbrains.com/issue/WEB-48085).

And I stopped looking after these two..